cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,214 @@
+        orderedProperties.entrySet().forEach(entry -> {
+            entry.setValue(entry.getValue().replaceAll("\\$\\{" + Pattern.quote(placeholder) + "\\}", value));
+        });
+    }
+
+    public KafkaConfiguration getDiff() {
+        return this.diff;
+    }
+
+    public boolean isDesiredPropertyDefaultValue(String key) {
+        Optional<ConfigEntry> entry = currentEntries.stream().filter(configEntry -> configEntry.name().equals(key)).findFirst();
+        if (entry.isPresent()) {
+            return entry.get().isDefault();
+        }
+        return false;
+    }
+
+    public boolean cannotBeUpdatedDynamically() {
+        if (diff == null) {
+            return false;
+        } else return diff.anyReadOnly(kafkaVersion)
+                || !diff.unknownConfigs(kafkaVersion).isEmpty()
+                || diff.containsListenersChange();
+    }
+
+    /**
+     * @return Map object which is used for dynamic configuration of kafka broker
+     */
+    public Map<ConfigResource, Collection<AlterConfigOp>> getUpdatedConfig() {
+        return updated;
+    }
+
+    private boolean isIgnorableProperty(String key) {
+        return IGNORABLE_PROPERTIES.matcher(key).matches();
+    }
+
+    /**
+     * Computes diff between two maps. Entries in IGNORABLE_PROPERTIES are skipped
+     * @return KafkaConfiguration containing all entries which were changed from current in desired configuration
+     */
+    public KafkaConfiguration computeDiff() {
+        Map<String, String> currentMap = new HashMap<>();
+        Map<String, String> difference = new HashMap<>();
+
+        Collection<AlterConfigOp> updatedCE = new ArrayList<>();
+
+        currentEntries.stream().forEach(e -> {
+            currentMap.put(e.name(), e.value());
+        });
+
+        Map<String, String> desiredMap = ModelUtils.configMap2Map(desired, "server.config");
+
+        fillPlaceholderValue(desiredMap, "STRIMZI_BROKER_ID", Integer.toString(brokerId));
+
+        JsonNode source = patchMapper().valueToTree(currentMap);
+        JsonNode target = patchMapper().valueToTree(desiredMap);
+        JsonNode jsonDiff = JsonDiff.asJson(source, target);
+
+        for (JsonNode d : jsonDiff) {
+            String pathValueWithoutSlash = d.get("path").asText().substring(1);
+            String pathValue = d.get("path").asText();
+
+            Optional<ConfigEntry> entry = currentEntries.stream().filter(e -> e.name().equals(pathValueWithoutSlash)).findFirst();
+
+            if (entry.isPresent()) {
+                if (d.get("op").asText().equals("remove")) {
+                    if (!entry.get().source().name().equals("DEFAULT_CONFIG")) {
+                        // we are deleting custom option
+                        difference.put(pathValueWithoutSlash, "deleted entry");
+                        updatedCE.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, entry.get().value()), AlterConfigOp.OpType.DELETE));
```

Comment:
We do that even if it's an ignorable path? If so it at least deserves a comment. If not, and we should guard with an ignorable check then we should add an else with trace logging. Also in that latter case it will make sense to pull out the ignorable check put outside this `if/else` chain, since all the branches would have the same logging for the ignorability case.

Reply:
I think custom property cannot be ignorable.